WWSTCERT-9096 Ct clamp driver wwsm

petewaddy-petewaddy opened the pull request:
Check all that apply

# Type of Change

- [x ] WWST Certification Request
     - If this is your first time contributing code:
          - [ x] I have reviewed the README.md file
          - [ x] I have reviewed the CODE_OF_CONDUCT.md file
          - [ x] I have signed the CLA
     - [x ] I plan on entering a WWST Certification Request or have entered a request through the WWST Certification console at developer.smartthings.com
- [ ] Bug fix
- [x ] New feature
- [ ] Refactor

# Checklist

- [ x] I have performed a self-review of my code
- [ x] I have commented my code in hard-to-understand areas
- [x ] I have verified my changes by testing with a device or have communicated a plan for testing
- [x ] I am adding new behavior, such as adding a sub-driver, and have added and run new unit tests to cover the new behavior

# Description of Change
add a sub driver to the standard edge driver for Zigbee Power meter to supply temperature  and battery percentage readings


# Summary of Completed Tests
Smartthings full 14 test test suite from the certification console, all 14 tests passed 

Test date: 13 November 2025 at 11:28:28
Test duration: 00:12:27
Session ID: WHJ1WLTHID

Test date: 13 November 2025 at 11:51:55
Test duration: 00:09:10
Session ID: 1DIHUP08CV

petewaddy-petewaddy committed:
no workspace files in repo

petewaddy-petewaddy committed:
start of using zigbee powerr meter as the main and then adding a sub driver for temperature, battery percentage

e

petewaddy-petewaddy committed:
first pass, after help from gemini.
battery seems to always be at 100% on both tested clamps, even throiugh one is as 98%

petewaddy-petewaddy committed:
Another attempt
Now calling the ct clamp sub driver - have temperature,battery,power meter and energy meter on the screen, last two are correct values, first two are not as yet

petewaddy-petewaddy committed:
first pass of what seems to be a fully working sub driver, packaged and deployed to the hub, using the standard smartthings zigbee power meter as the base and adding as a sub driver the temperature and battery level

petewaddy-petewaddy committed:
Changes to copied in values for temperature and battery percentage

petewaddy-petewaddy committed:
remove reference to VS code workspace

petewaddy-petewaddy committed:
set back to origin code

petewaddy-petewaddy committed:
no message

petewaddy-petewaddy committed:
cr lf

petewaddy-petewaddy committed:
WWSTCERT-9096
PR #2552

indent code under the statement to the right   as per pull request comment

pegor-karoglanian reviewed (approved):
lgtm

greens commented on drivers/SmartThings/zigbee-power-meter/src/chameleon/init.lua:
missing copyright/license statement

greens commented on drivers/SmartThings/zigbee-power-meter/fingerprints.yml:
```diff
@@ -28,10 +28,36 @@ zigbeeManufacturer:
+        - 0x0003 #Identify
+    deviceProfileName: ct-clamp    
+
+    
```
Please just include the manufacturer-specific fingerprint

greens commented on .gitignore:
changes to this file should be omitted

greens commented:
```
Checking drivers/SmartThings/zigbee-power-meter/src/chameleon/init.lua 2 warnings

    drivers/SmartThings/zigbee-power-meter/src/chameleon/init.lua:2:7: (W211) unused variable cluster_base
    drivers/SmartThings/zigbee-power-meter/src/chameleon/init.lua:3:7: (W211) unused variable data_types
```

petewaddy-petewaddy committed:
ct-clamp-driver-wwsm
rename file from ct-clamp.yml to power-energy-battery-temperature.yml
add copyright information

petewaddy-petewaddy committed:
ct_clamp-driver_wwsm
removed unused variables cluster_base and data_types

petewaddy-petewaddy committed:
ct_clamp_river_wwsm
add changed name into yaml file
remove non manufacturer specific items

greens reviewed (commented):
Please address the remaining comments.

wkhenon commented on drivers/SmartThings/zigbee-power-meter/fingerprints.yml:
```diff
@@ -28,10 +28,18 @@ zigbeeManufacturer:
     deviceProfileName: power-meter
+
+
+    
```
```suggestion
```

ianlovatt replied:
made suggested change and removed newlines

wkhenon commented on drivers/SmartThings/zigbee-power-meter/src/chameleon/init.lua:
```diff
@@ -0,0 +1,99 @@
+  {model = "CT101xxxx" }
+}
+
+-- temperature: 0.5C, battery level remaining: 1%
```
Is this comment still needed?

petewaddy-petewaddy committed:
ct_clap_driver_wwsm
Added longer copyright at top of our init.lua file
removed comment about tempeature and battery
removed battery level handler to main init.lua file under capabilities.battery, temperature syayed as its cluster 0x0402 not the normal temperature cluster and didn't update the temperature

petewaddy-petewaddy committed:
removed as causing error in the HUB log

petewaddy-petewaddy committed:
ct_clamp_driver_wwsm
removed 3 lines as requested in comments on the pull request

petewaddy-petewaddy committed:
ct_clamp_driver_wwsm
remove unintentional extra line in .gitignore

petewaddy-petewaddy committed:
ct_clamp_driver_wwsm
add new line, that was erased by accicent in these two files

petewaddy-petewaddy committed:
ct_clamp_driver_wwsm
trying to insert a new line at the end of the file to remove the red circle in github, when the file is checked in

petewaddy-petewaddy committed:
ct_clamp_driver_wwsm
remove battery percentage config, and use defaults
remove battery_level_handler as usinf the capabilites function
remove log function in temperature handler
adjust spacing in is_chameleon_ct_clamp function

petewaddy-petewaddy committed:
ct_clamp_driver_wwsm
after a rebase add in the chameleon entry into the sub_drivers.lua file

petewaddy-petewaddy committed:
ct_clamp_driver_wwsm
after a rebase add in the chameleon entry into the sub_drivers.lua file
add in the new files can_handle.lua and fingerprints.lua, for the new type of sub driver architecture

petewaddy-petewaddy committed:
Merge branch 'main' into ct_clamp_driver_wwsm

greens commented on drivers/SmartThings/zigbee-power-meter/src/chameleon/init.lua:
```diff
@@ -0,0 +1,73 @@
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
```
please use the shorter copyright/license statement

greens reviewed (changes requested):
Since you're using a unique attribute, please write some unit tests to verify the behavior. We prefer code coverage to be >90%.

KKlimczukS commented on drivers/SmartThings/zigbee-power-meter/profiles/power-energy-battery-temperature.yml:
```diff
@@ -0,0 +1,19 @@
+  - id: temperatureMeasurement
+    version: 1
+  categories:
+  - name: CurbPowerMeter
```
```suggestion
  - name: PowerMeasurementSensor
```

KKlimczukS replied:
Don't You want to add support for powerConsumptionReport capability ?

KKlimczukS reviewed (commented):
Please consider using this category: PowerMeasurementSensor
https://developer.smartthings.com/docs/certification/test-suite#supported-device-types
